src/server.c

```diff
@@ -3628,9 +3634,9 @@ int processCommand(client *c) {
      * 1) The sender of this command is our master.
      * 2) The command has no key arguments. */
     if (server.cluster_enabled &&
-        !(c->flags & CLIENT_MASTER) &&
+        !mustObeyClient(c) &&
```

Comment:
Doesn't this imply a cluster node can now load an AOF file that is incompatible with its slot configuration? I think we'd want to fail in this case.

Reply:
well, maybe, but as Zhao mentioned, AOF doesn't actually reach here.
i'll trim this part from the top comment.

Reply:
This still seems problematic anyways, if a replica has data in slots it doesn't think it owns, when it get's promoted it won't be able to serve them and just have stuck data. This seems more like an error.

Reply:
so what do you suggest? remove that `if`, let it error and possibly panic?

here's where that was added: d3b4662347a1099d90339328ef6df7bc939cf42a
seems very on purpose..

Reply:
I would vote panic. Salvatore was never one to do all that much validation.

Reply:
ok, so i guess that makes this change part of your PR then.
i.e. my PR doesn't have any panic, it's mostly about 3 things:
1. allowing commands to run instead of rejecting them (if they come from master or AOF)
2. rejecting commands on replicas, if the client isn't the master client.
3. deleting dead code

i didn't have any changes to add rejection of commands when the client is the master, and if we wanna do that and rely on the panic, then it should be in your PR.
agree?